packages/hardhat/scripts/runHardhatDeployWithPK.ts

```diff
@@ -0,0 +1,54 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+import { Wallet } from "ethers";
+import password from "@inquirer/password";
+import { spawn } from "child_process";
+
+/**
+ * Unencrypts the private key and runs the hardhat deploy command
+ */
+async function main() {
+  const isNetworkProvided = process.argv.slice(2).includes("--network");
+
+  if (!isNetworkProvided) {
```

Comment:
Maybe we want to check that if network is specified, it's different that "hardhat"

Reply:
Yes, it will be a safe check.